src/index.ts

```diff
@@ -40,14 +44,55 @@ export function runInContext(
   const program = parse(code, context)
   if (program) {
     if (theOptions.isNativeRunnable) {
+      let transpiled
+      let sourceMapJson
+      let lastStatementSourceMapJson
       try {
+        const temp = transpile(program, context.contextId)
+        // some issues with formatting and semicolons and tslint so
+        transpiled = temp[0]
+        sourceMapJson = temp[1]
+        lastStatementSourceMapJson = temp[2]
         return Promise.resolve({
           status: 'finished',
-          value: sandboxedEval(generate(transpile(program, context.contextId)))
+          value: sandboxedEval(transpiled)
         } as Result)
       } catch (error) {
-        context.errors.push(new ExceptionError(error, UNKNOWN_LOCATION))
-        return Promise.resolve({ status: 'error' } as Result)
+        if (error instanceof RuntimeSourceError) {
+          context.errors.push(error)
+          return resolvedErrorPromise
+        }
+        const errorStack = error.stack
+        const match = /<anonymous>:(\d+):(\d+)/.exec(errorStack)
+        if (
+          match === null ||
+          sourceMapJson === undefined ||
+          lastStatementSourceMapJson === undefined
+        ) {
+          context.errors.push(new ExceptionError(error, UNKNOWN_LOCATION))
+          return resolvedErrorPromise
+        }
+        const line = Number(match![1])
+        const column = Number(match![2])
```

Comment:
Seems like you're parsing the stacktrace, did you consider using a library? A quick google search yields StackTrace.JS, but I'm not sure if overkill.

Reply:
It's just parsing a single line, nothing too advanced yet.